MaterialLib/CMakeLists.txt

```diff
@@ -16,6 +16,8 @@ append_source_files(SOURCES Fluid/WaterVaporProperties)
 
 append_source_files(SOURCES MPL)
 append_source_files(SOURCES MPL/Properties)
+append_source_files(SOURCES MPL/Properties/CapillaryPressureModel)
```

Comment:
This is the new folder for the capillary pressure models. The saturation models shouldn't be here, isn't it?

Reply:
Physically, there is no saturation models, which in the source code are the inverse of  capillary pressure (saturation) model. I would prefer to put all  capillary pressure saturation model and their inverse models in this directory. The directory name would be CapillaryPressureSaturation.

Reply:
As far as I understand this discussion, the question is whether a new directory is called 'capillary...' or 'saturation...'. From Dimas perspective, there should be two directories, one for capillary pressure models and one for saturation models. From Wenqings perspective, there is just one type of model, which returns either capillary pressure or saturation depending on the process. 

Personally, I would also prefer to have just one implementation for each model, which is then inverted or not depending on the process requirements. In that case, I agree to Dima that we need to make sure that the functions are always invertible (we need to regard cut-off pressures, etc. in that case). To avoid confusion, I suggest to rename the new directory to 'characteristic_curves', since this is the proper name for this type of model and it shows that those functions can be used in any direction.

Reply:
@Scinopode 
1. For the invertible, Dima suggested two classes for one model, one for direct and another one for invertible. I thought that it was good point because of the meaning of  the returned value of `...::value()` is consistent with the property name. My previous implementation is adding inverse_value to Property (see my closed PR #2854). Which one is better? I prefer to take Dima's suggection and have already implemented some.

2. For the name of new directory, since I want to moved all capillary pressure  saturation model into it, `characteristic_curves` seems can not represents those physical properties properly.

Reply:
As a simple suggestion, we could do the generalization (*e.g.* moving files around, creating inheritance trees, *etc.*) after the `CapillaryPressureVanGenuchten` model is added.